api/src/main/java/io/opencensus/tags/unsafe/ContextUtils.java

```diff
@@ -41,10 +42,37 @@ private ContextUtils() {}
    * {@link io.grpc.Context}.
    *
    * @since 0.8
+   * @deprecated from API since 0.21. Use {@link #withValue(Context, TagContext)} and {@link
+   *     #getValue(Context)} instead.
    */
+  // TODO(songy23): make this private once gRPC migrates to use the alternative APIs.
+  @Deprecated
   public static final Context.Key<TagContext> TAG_CONTEXT_KEY =
       Context.keyWithDefault("opencensus-tag-context-key", EMPTY_TAG_CONTEXT);
 
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param context the parent {@code Context}.
+   * @param tagContext the value to be set.
+   * @return a new context with the given value set.
+   * @since 0.21
+   */
+  public static Context withValue(Context context, TagContext tagContext) {
```

Comment:
Shouldn't you check whether tagContext is null? Or annotate as nullable?

Reply:
No, because the key is initialized with a default value.

Reply:
Reverted, though I'm wondering when we would want to set `TagContext` or `Span` to null in a context.

Reply:
Agreed that the default value does mean the tagContext can be null but still not clear to me that there is a good use-case for this.

If there is, shouldn't it be annotated Nullable?

Reply:
I added the Nullable annotation to the context value. The new API still returns non-null value though.